src/SendCmioResponse.sol

```diff
@@ -0,0 +1,88 @@
+// Copyright Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+/// @title SendCmioResponse
+/// @notice Sends a CMIO response
+//:#include macro.pp
+/// DEV_COMMENT(templates/SendCmioResponse.sol.template)
+
+pragma solidity ^0.8.0;
+
+import "./EmulatorCompat.sol";
+
+library SendCmioResponse {
+    using Memory for uint64;
+    using AccessLogs for AccessLogs.Context;
+
+    // START OF AUTO-GENERATED CODE
+
+    function sendCmioResponse(
+        AccessLogs.Context memory a,
+        uint16 reason,
+        bytes32 dataHash,
+        uint32 dataLength
+    ) internal pure {
+        if (!EmulatorCompat.readIflagsY(a)) {
+            EmulatorCompat.throwRuntimeError(a, "iflags.Y is not set");
+        }
+        // A zero length data is a valid response. We just skip writing to the rx buffer.
+        if (dataLength > 0) {
+            // Find the write length: the smallest power of 2 that is >= dataLength and >= tree leaf size
+            uint32 writeLengthLog2Size = EmulatorCompat.uint32Log2(dataLength);
```

Comment:
I think using this `uint32Log2` primitive forces us to do more accounting later. In Dave consensus, we use a `log2clp` approach. I suggest we use the same approach here. Take a look at this [comment](https://github.com/cartesi/dave/pull/81#discussion_r1809244557) outlining the solution, and at the final [implementation](https://github.com/cartesi/dave/blob/84fd797a68fe78ae604413b8514063db0d3a02b3/cartesi-rollups/contracts/src/Math.sol#L61).

I think it's easier than this
```
writeLengthLog2Size = floor(log2(dataLength));
writeLengthLog2Size += (1 << f) < dataLength ? 1 : 0
```

Reply:
With `log2clp`, then it becomes just:

```
// suggestion with log2clp
writeLengthLog2Size = log2clp(dataLength);
```

Reply:
`log2_clp2` in Lua:

```lua
local function log2_clp2(x)
  assert(math.ult(0, x))
  return 64 - clz(x - 1)
end
```